include/highfive/bits/H5ReadWrite_misc.hpp

```diff
@@ -28,6 +28,7 @@ struct BufferInfo {
     const bool is_fixed_len_string;
     const size_t n_dimensions;
     const DataType data_type;
+    bool warning_thrown = false;
```

Comment:
Why inside `BufferInfo`

Reply:
It is there that the the warning is generated. I'm not sure what you have in mind?

Reply:
I was going to comment on this earlier… Maybe `BufferInfo::getDataType` could get a provenance `string` parameter that would appended to the warning?

This would save this member variable, and avoid having to do checks outside the call, even if passing a parameter is not that pretty.

Reply:
(Because I don't think HighFive stores the name as member?)

Reply:
If overhead is a concern, I guess the name computing function can be passed in directly/via a lambda-callback?

Reply:
With passing a lambda function I can very well live ! Please check the updated code